subsys/net/ip/tcp.c

```diff
@@ -2312,12 +2364,21 @@ int net_tcp_listen(struct net_context *context)
 
 int net_tcp_update_recv_wnd(struct net_context *context, int32_t delta)
 {
-	if (!context->tcp) {
+	struct tcp *conn = context->tcp;
+	int ret;
+
+	if (!conn) {
 		NET_ERR("context->tcp == NULL");
 		return -EPROTOTYPE;
 	}
 
-	return tcp_update_recv_wnd((struct tcp *)context->tcp, delta);
+	k_mutex_lock(&conn->lock, K_FOREVER);
+
```

Comment:
why those empty lines around the `tcp_update_recv_wnd()`?

Reply:
Personally, I think it's easier to analyze then, when you separate the actual logic from thread-safety protection. But it also follows the general convention used in this file: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/tcp.c#L530.
Not sure thou if it's against the general coding guidelines (at least compliance doesn't complain).